File: chat_service/src/main/java/com/ChatService/model/doctor.java
package com.ChatService.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor

public class doctor {
    @Id
    private String email;
    private String firstname;
    private String lastname;
    private String dob;
    private String password;
    private String mobilenumber;
    private String certificate;
}

File: Frontend/lib/doctor.dart
class Doctor {
  String dob;
  String email;
  String firstname;
  String lastname;
  String mobilenumber;
  String password;
  String certificate;

  Doctor(this.dob, this.email, this.firstname, this.lastname, this.mobilenumber,
      this.password, this.certificate);

  Map toJson() => {
        'dob': dob,
        'email': email,
        'firstname': firstname,
        'lastname': lastname,
        'mobilenumber': mobilenumber,
        'password': password,
        'certificate': certificate
      };

  factory Doctor.fromJson(dynamic json) {
    return Doctor(
        json['dob'] as String,
        json['email'] as String,
        json['firstname'] as String,
        json['lastname'] as String,
        json['mobilenumber'] as String,
        json['password'] as String,
        json['certificate'] as String);
  }
}
